### PR TITLE
One Response Model + Credentials as a Parameter + Request Models + Dockerfile

### PR DESCRIPTION
**Pull request by kangwork**
**Description**
Please refer to the commit messages below for more detailed descriptions. ;)
Note that all the changes made have been tested fully.

- **[Aggregated List]**
    - Studied and made the use of `aggregated_list` clearer and more efficient

- **[ABC]**
    - Studied ABC and made the Collector class as an abstract class
    - Made some simplifications

- **[Resource Model]**
    - Worked on consistency

- **[Response Model] ⭐** 
    - Defined a single response model that can be used in any APIs
    - Applied to all APIs and completed testing

- **[Credentials as Parameter] ⭐** 
    - Let users input the credentials information if wanted (otherwise find a pre-defined environmental variable if exists)
    - Worked on consistency + efficiency

- **[Style, Black Formatter, Makefile, Exception, etc.]**
    - See below

- **[Request Models] ⭐** 
    - Defined request models based on the updated specifications
    - Changed some credentials gathering logic + types accordingly

- **[Dockerfile] ⭐** 
    - Wrote a dockerfile
    - Changed some imports + type notations to incorporate the 3.8 compatibility

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -38,9 +38,9 @@ def read_root(request: Request):
 ### 2-5. All Three at Once
 # 2-5-1. A route to list all resources in a project
 # Example use: http://localhost/all-resources
-@app.get("/all-resources")
+@app.post("/all-resources", response_model=APIResponse)
 @func_error_handler_decorator(logger=logger, is_api=True)
-def list_all_resources():
+def list_all_resources(credentials=Depends(get_credentials)):
```
Comment (ImMin5) [inline src/main.py]: 사용자가 secret_data(dict) parameter로 입력 할 수 있게 변경 되면 좋을 것 같습니다.

Comment (ImMin5) [inline src/main.py]: 제가 예상 했던 모습은 아래와 같습니다!
image

File: src/models/response.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+class APIResponse(BaseModel):
+    data: str | list | dict[str, list]
+    length: int = -1
+    message: str = "Thanks for using the Mini Google Cloud Collector."
+
+    def __init__(self, **data):
+        super().__init__(**data, length=self.get_length(data, data))
+
+    def get_length(cls, v, values) -> int:
```
Comment (ImMin5) [inline src/models/response.py]: cls variable을 사용하신 이유가 있을 까요?

File: src/models/response.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+from pydantic import BaseModel
+
+
+class APIResponse(BaseModel):
```
Comment (ImMin5) [inline src/models/response.py:5]: 미팅때 APIResponse class에 대해서 간단하게 설명 부탁드립니다!

**Review by ImMin5 — CHANGES_REQUESTED** [review]

File: src/main.py (hunk 1 above)
Comment (kangwork) [reply src/main.py]: 확인하였습니다! 피드백 감사합니다!

File: src/models/response.py (hunk 3 above)
Comment (kangwork) [reply src/models/response.py:5]: 넵!

File: src/main.py (hunk 1 above)
Comment (kangwork) [reply src/main.py]: 넵! 그렇게 수정하도록 하겠습니다!

File: src/models/response.py (hunk 2 above)
Comment (kangwork) [reply src/models/response.py]: 실수입니다.. 디자인 바꾸고 파라미터 변경을 놓쳤습니다.
덕분에 다시 공부할 수 있었습니다. 알려주셔서 감사합니다!

**Pull request merged**